File: Faza6/OnlineGamesShop/src/main/java/com/rich/onlinegamesshop/model/reports/YearlyGameStatisticsReport.java
package com.rich.onlinegamesshop.model.reports;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class YearlyGameStatisticsReport {
    private String title;
    private BigDecimal yearNumber;
    private long totalPlatforms;
    private BigDecimal averageRating;
    private BigDecimal totalSales;
}

File: Faza6/FrontEnd/Angular/src/app/model/Reports/YearlyGameStatisticsReport.ts
import Big from "big.js";

export class YearlyGameStatisticsReport{
  title:string;
  yearNumber:Big;
  totalPlatforms:number;
  averageRating:Big;
  totalSales:Big;
  constructor(
    title:string,
    yearNumber:Big,
    totalPlatforms:number,
    averageRating:Big,
    totalSales:Big,
  ){
    this.title=title;
    this.yearNumber=yearNumber;
    this.totalPlatforms=totalPlatforms;
    this.averageRating=averageRating;
    this.totalSales=totalSales;
  }
}
